### PR TITLE
Add Details To Plate result

### PR DESCRIPTION
**Pull request by amirzenoozi**
Add details to the info object to have a dynamic template.
I need to use each part of the template separately in my components so I decide to add new fields in the info object.
I think it can be helpful for others.

### CODE REVIEW

File: src/modules/numberplate/types.skip.ts [hunk 1]
```diff
@@ -31,6 +31,26 @@ export enum PlateTypes {
  */
 export type PlateResultApiTypeString = keyof typeof PlateTypes;
 
+/**
+ *
+ * @category Plate
+ */
+export interface PlateResultApiCarDetailsObject {
+	first_two_digits: string;
```
Comment (ali-master) [inline src/modules/numberplate/types.skip.ts]: All the Keys should follow the `CamelCase` model.

File: src/modules/numberplate/types.skip.ts [hunk 2]
```diff
@@ -31,6 +31,26 @@ export enum PlateTypes {
  */
 export type PlateResultApiTypeString = keyof typeof PlateTypes;
 
+/**
+ *
+ * @category Plate
+ */
+export interface PlateResultApiCarDetailsObject {
```
Comment (ali-master) [inline src/modules/numberplate/types.skip.ts]: Replace `PlateResultApiCarDetailsObject` to `PlateResultDetailModel`

File: src/modules/numberplate/index.ts [hunk 3]
```diff
@@ -75,10 +77,17 @@ export function carHandler(plate: NormalizedPlate): PlateResultApi {
 
 	const province = plateDataset.Car[provinceCode];
 	const category = plate.char ? plateDataset.Category[plate.char] : null;
+	const details: PlateResultApiCarDetailsObject = {
+		first_two_digits: plate.numbers.slice(0, 2),
+		plate_character: plate.char ? plate.char : null,
```
Comment (ali-master) [inline src/modules/numberplate/index.ts]: Replace `plate.char ? plate.char : null` with `plate.char ?? : null`

File: src/modules/numberplate/index.ts [hunk 4]
```diff
@@ -89,11 +98,16 @@ export function motorcycleHandler(plate: NormalizedPlate): PlateResultApi {
 	const template = `${provinceCode}-${plate.numbers.slice(3)}`;
 
 	const province = plateDataset.Motorcycle[provinceCode];
+	const details: PlateResultApiMotorcycleDetailsObject = {
```
Comment (ali-master) [inline src/modules/numberplate/index.ts]: Replace `PlateResultApiMotorcycleDetailsObject` with `PlateResultApiMotorcycleDetailModel`

**Review by ali-master — COMMENTED** [review]
Please squash your commits and follow the Conventional Commits messaging rules. All commits are wrong and is not acceptable.

**Comment by ali-master** [comment]
@amirzenoozi Your commits message is wrong and is not acceptable.

**Comment by amirzenoozi** [comment]
@ali-master 
I will fix it now.
I checked the previous commits in this repo and squash all previous commits with you your format.

**Comment by ali-master** [comment]
@amirzenoozi Jan, We want to release a new version and We would like to see your changes in this release, may you complete your PR in up to 3 more days?

**Comment by amirzenoozi** [comment]
@ali-master Jan, Should I do something else ?

**Comment by ali-master** [comment]
If you solved all issues in the review, please mark them as ```resolve conversation```.

**Comment by amirzenoozi** [comment]
I forgot to mark them as `resolve conversation`, it's done.

**Comment by ali-master** [comment]
Thanks, Please update README and let me know to merge your PR.

**Comment by ali-master** [comment]
Merged, thanks for contributing.

I have an idea, we should write a function that gives ```provinceCode``` and returns the Province's name in English and Persian. what do you think? Do you have time to write this function?

**Comment by ali-master** [comment]
@amirzenoozi 

Bad news, Plate tests will be failed because of the details property which you've added. Can you create another PR and fix this problem ASAP?

**Comment by ali-master** [comment]
@all-contributors please add @amirzenoozi  for code, test and ideas.

**Comment by amirzenoozi** [comment]
@ali-master first I just work on plate tests to fix it ASAP, second about the function that should return the `provinceCode`, you want this function for this release? and I have a simple question about the categories, why we don't have `Taxi` as a category?

**Comment by amirzenoozi** [comment]
@ali-master 
I think the test is fixed but, when I run the `numberplate` test I got an error on this line `plate.char ?? : null` what should I do?

**Comment by ali-master** [comment]
> @ali-master first I just work on plate tests to fix it ASAP, second about the function that should return the `provinceCode`, you want this function for this release? and I have a simple question about the categories, why we don't have `Taxi` as a category?

Thanks for the first issue which you want to fix it. 
My suggestion is that it would be better If had this function in this release.
About the second issue you've raised, I have no idea and please connect to @moeindev and tell him the story because He wrote this function and he knows all details.

Please join our Slack workspace for contacting @mediv0. he is there.

https://join.slack.com/t/persiantools/shared_invite/zt-ss9m76wy-oA4vvYAXN3FrKNZZ44LAuQ

**Comment by mediv0** [comment]
> @ali-master
> I think the test is fixed but, when I run the `numberplate` test I got an error on this line `plate.char ?? : null` what should I do?

because  `plate.char ?? : null` is not a valid javascript syntaxt. you probably meant to write this: `plate.char ?? null`

**Comment by mediv0** [comment]
> @ali-master first I just work on plate tests to fix it ASAP, second about the function that should return the `provinceCode`, you want this function for this release? and I have a simple question about the categories, why we don't have `Taxi` as a category?

feel free to add taxi code to cateogry list

**Comment by mediv0** [comment]
also in
```js
export interface PlateResultDetailModel {
	firstTwoDigits: string;
	plateCharacter: string;
	nextThreeDigits: string;
	provinceCode: string;
}
```

you need to change this interface to: 
```js
export interface PlateResultDetailModel {
	firstTwoDigits: string;
	plateCharacter: string | null;
	nextThreeDigits: string;
	provinceCode: number;
}
```

or for refactor you can also replace `plateCharacter: plate.char ?? null` with `plateCharacter: category`

**Comment by amirzenoozi** [comment]
@mediv0 It's fixed thanks for your help, now I have a little problem with `pre-commit` the `prettier:ci` task returns me this error: 
```
No files matching the pattern were found: "'src/**/*.{ts,json}'".
```

**Comment by mediv0** [comment]
> @mediv0 It's fixed thanks for your help, now I have a little problem with `pre-commit` the `prettier:ci` task returns me this error:
> 
> ```
> No files matching the pattern were found: "'src/**/*.{ts,json}'".
> ```

just remove single quotes around `src/**/*.{ts,json}` , works on windows

``` "prettier:ci": "prettier --list-different src/**/*.{ts,json} --config .prettierrc.js", ```

**Comment by amirzenoozi** [comment]
@ali-master @mediv0 I fix the problem and all the `platenumber` tests passed but because of some of the other tests that will fail I can't push anything, what should I do?

**Comment by mediv0** [comment]
> @ali-master @mediv0 I fix the problem and all the `platenumber` tests passed but because of some of the other tests that will fail I can't push anything, what should I do?

```git commit --no-verify and git push --no-verify```  should allow you to commit i think

**Pull request merged**